content/08-mongodb/10-mongodb-transactions.mdx

```diff
@@ -0,0 +1,246 @@
+
+For the sake of demonstration, we'll walk through the creation of a transaction via the MongoDB shell. We'll work with an example database and collection to help conceptualize how transactions work in practice.
+
+**Note**: If a transactions session runs for more than 60 seconds, MongoDB will automatically abort the operation. This occurs because transactions are typically conducted in applications that work automatically rather than a person issuing commands in the shell. The following steps are for conceptualizing a transaction session from start to finish.  
+
+To begin, we have a simple database called `literature` with a collection called `authors`. Running a quick `find()` query, we can see the document structure with the author name and a title of theirs.
+
+```
+db.authors.find()
+```
+```
+[
+  {
+    _id: ObjectId("620397dd4b871fc65c193106"),
+    first_name: 'James',
+    last_name: 'Joyce',
+    title: 'Ulysses'
+  },
+  {
+    _id: ObjectId("620398016ed0bb9e23785973"),
+    first_name: 'William',
+    last_name: 'Gibson',
+    title: 'Neuromancer'
+  },
+  {
+    _id: ObjectId("6203981d6ed0bb9e23785974"),
+    first_name: 'George',
+    last_name: 'Orwell',
+    title: 'Homage to Catalonia'
+  },
+  {
+    _id: ObjectId("620398516ed0bb9e23785975"),
+    first_name: 'James',
+    last_name: 'Baldwin',
+    title: 'The Fire Next Time'
+  }
+]
+```
+Now that we have our database, we can show the steps to work with a transaction directly in the shell and demonstrate how a transaction being conducted in one session may not be detectable to an outside source.
+
+First, we need to create a session, same as you would when using the API.
+
+```
+var session = db.getMongo().startSession()
+```
+This newly created `session` variable will store the session object.
+
+The next step is starting the transaction by calling the `startTransaction()` method:
+
+```
+session.startTransaction( 
+	{ "readConcern": { "level": "snapshot" },
+	  "writeConcern": { "w": "majority } 
+	 } 
+)
+```
+The `startTransaction()` method has two options, `readConcern` and `writeConcern`. You can read in more detail about these options in [MongoDB's documentation](https://docs.mongodb.com/manual/core/transactions/#read-concern-write-concern-read-preference). We set the `readConcern` `level` to [`snapshot`](https://docs.mongodb.com/manual/core/transactions/#-snapshot-) which returns data from a snapshot of majority committed data *if* the transaction commits with `writeConcern` ["majority"](https://docs.mongodb.com/manual/core/transactions/#w---majority-). When you commit with `w: "majority"` write concern, transaction level `"snapshot"` read concern guarantees that operations have a synchronized snapshot of majority-committed data and can be thought of as good defaults if no specific requirements.
```

Comment:
It may just be me, but this paragraph was pretty hard for me to parse towards the second half.  I think maybe it's that the last two sentences have a lot of clauses and they seem to be circling around the same idea.  Just a heads up.